basin3d/core/catalog.py

```diff
@@ -307,168 +340,281 @@ def _process_plugin_attr_mapping(self, plugin, filename: str, datasource: DataSo
                 if not valid_type_vocab:
                     continue
 
-                attr_mapping = AttributeMapping(
+                attr_mapping = basin3d.core.models.AttributeMapping(
                     attr_type=attr_type,
                     basin3d_vocab=basin3d_vocab,
                     basin3d_desc=basin3d_desc,
                     datasource_vocab=datasource_vocab,
                     datasource_desc=datasource_desc,
-                    datasource=datasource)
+                    datasource=basin3d.core.models.DataSource(id=datasource.id, name=datasource.name,
+                                                              location=datasource.location,
+                                                              id_prefix=datasource.id_prefix))
```

Comment:
Why rebuild the datasource object here? i.e., instead of using the object that is passed in?

Reply:
Good catch